### PR TITLE
Creates Subworkflows

### PR DESCRIPTION
**Pull request by BWMac**
This PR creates a subworkflow structure for Challenge Nextflow workflows. It implements subworkflows that can be invoked by using the `entry` parameter to run the workflow locally:
```
nextflow run main.nf -entry MODEL_TO_DATA_CHALLENGE -profile local
```
or on Tower using the `Workflow entry name` field under `Advanced options`.

General structure/philosophy of the repository:
- Modules (process definitions) remain in the `modules` folder. Module internal logic should not be changed unless absolutely necessary in order to avoid breaking any of the subworkflows (we will need a testing structure to avoid breaking changes in the future). If different processes are needed as we support new challenges, new modules will need to be contributed.
- Subworkflows are located in the `subworkflows` folder. Each file should contain a named workflow that is created by stringing together various modules. A subworkflow will be created for each challenge, and the workflow will be named clearly after that challenge.
- All subworkflows are pulled into `main.nf` so that they can be executed when the pipeline runs. The named workflows defined in the subworkflow are wrapped in a new named workflow. This final workflow name will serve as the entrypoint for the workflow when that challenge is run.

Both subworkflows have been tested locally and on Tower.
- [Model To Data Tower run](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/4U9QuhZEJEK5VR)
- [Data To Model Tower run](https://tower-dev.sagebionetworks.org/orgs/Sage-Bionetworks/workspaces/example-dev-project/watch/YGD7E2woEwFWQ)

### CODE REVIEW

File: img/data_to_model_dag.png
Comment (thomasyu888) [inline img/data_to_model_dag.png:1]: Nit: Explore this - https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/ so we don't have to maintain a separate png

File: subworkflows/DATA_TO_MODEL.nf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+params.view_id = "syn52576179"
+
+// import modules
+include { SYNAPSE_STAGE } from '../modules/synapse_stage.nf'
+include { GET_SUBMISSIONS } from '../modules/get_submissions.nf'
+include { UPDATE_SUBMISSION_STATUS as UPDATE_SUBMISSION_STATUS_BEFORE_RUN } from '../modules/update_submission_status.nf'
+include { DOWNLOAD_SUBMISSION } from '../modules/download_submission.nf'
+include { UPDATE_SUBMISSION_STATUS as UPDATE_SUBMISSION_STATUS_AFTER_VALIDATE } from '../modules/update_submission_status.nf'
+include { UPDATE_SUBMISSION_STATUS as UPDATE_SUBMISSION_STATUS_AFTER_SCORE } from '../modules/update_submission_status.nf'
+include { VALIDATE } from '../modules/validate.nf'
```
Comment (thomasyu888) [inline subworkflows/DATA_TO_MODEL.nf:15]: Per our discussion, we will have different validate.nf modules per challenge?

**Review by thomasyu888 — APPROVED** [review]
💯 Nice!  I'm going to pre-approve, but just some comments

File: subworkflows/DATA_TO_MODEL.nf (hunk 1 above)
Comment (BWMac) [reply subworkflows/DATA_TO_MODEL.nf:15]: my hope is to have modules be as multi-purpose as possible by parameterizing the validation/scoring scripts and containers (as you mentioned in the other PR). I was imagining having all of these parameters wrapped up in a unique configuration for each challenge.

So basically subworkflows are on the _challenge type_ level, and configurations are on the _individual challenge_ level, but I think we need to see how different the needs will be for each challenge.

Comment (thomasyu888) [reply subworkflows/DATA_TO_MODEL.nf:15]: Got it. I think what we should do is keep this in the back of our heads and modify the workflows as we add challenges to this monorepo.

File: img/data_to_model_dag.png
Comment (BWMac) [reply img/data_to_model_dag.png:1]: implemented this diagramming method in my latest commit.

**Pull request merged**